mlflow/models/evaluation/base.py

```diff
@@ -957,6 +951,26 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
     raise ModelValidationFailedException(message=os.linesep.join(failure_messages))
 
 
+def _convert_data_to_mlflow_dataset(data, targets=None, dataset_path=None):
+    """Convert input data to mlflow dataset."""
+    if "pyspark" in sys.modules:
+        from pyspark.sql import DataFrame as SparkDataFrame
+
+    if isinstance(data, Dataset):
+        return data
+    elif isinstance(data, pd.DataFrame):
+        return mlflow.data.from_pandas(df=data, targets=targets, source=dataset_path)
+    elif "pyspark" in sys.modules and isinstance(data, SparkDataFrame):
+        return mlflow.data.from_spark(df=data, targets=targets, path=dataset_path)
+    elif isinstance(data, np.ndarray):
+        return mlflow.data.from_numpy(data, targets=targets, source=dataset_path)
+    else:
```

Comment:
in the [mlflow.evaluate](https://mlflow.org/docs/latest/python_api/mlflow.html?highlight=mlflow%20evaluate#mlflow.evaluate) docs we state that `data` must be 

> A numpy array or list of evaluation features, excluding labels.
> A Pandas DataFrame or Spark DataFrame, containing evaluation features and labels. If feature_names argument not specified, all columns are regarded as feature columns. Otherwise, only column names present in feature_names are regarded as feature columns. If it is Spark DataFrame, only the first 10000 rows in the Spark DataFrame will be used as evaluation data.
> A :py:class`mlflow.data.dataset.Dataset` instance containing evaluation features and labels.

can we confirm that the "list of evaluation features" case is properly handled?

Reply:
Sorry I missed this type, added conversion code and a unit test case.